packages/tss-client/src/client.ts

```diff
@@ -473,3 +392,84 @@ export class Client {
   }
 }
+
+// TODO: create namespace for globals
+if (globalThis.tss_clients === undefined) {
+  // Cleanup leads to memory leaks with just an object. Should use a map instead.
+  // TODO: This should be singular
+  globalThis.tss_clients = new Map();
+}
+
+if (globalThis.js_read_msg === undefined) {
+  globalThis.js_read_msg = async function (session: string, self_index: number, party: number, msg_type: string) {
+    const tss_client = globalThis.tss_clients.get(session) as Client;
+    tss_client.log(`reading msg, ${msg_type}`);
+    if (msg_type === "ga1_worker_support") {
+      return "unsupported";
+    }
+    const mm = tss_client.msgQueue.find((m) => m.sender === party && m.recipient === self_index && m.msg_type === msg_type);
+    if (!mm) {
+      // It is very important that this promise can reject, since it is passed through to dkls library and awaited internally. If it cannot reject and a message is lost,
+      // it will never resolve and hang indefinitely with no possibility of recovery.
+      return new Promise((resolve, reject) => {
+        let counter = 0;
+        const timer = setInterval(() => {
+          const found = tss_client.msgQueue.find((m) => m.sender === party && m.recipient === self_index && m.msg_type === msg_type);
+          if (found !== undefined) {
+            clearInterval(timer);
+            resolve(found.msg_data);
+          }
+          if (counter >= 1000) {
+            clearInterval(timer);
+            // TODO Fix wasm to handle error objects properly and then reject
+            // with Error instead of string.
+            //
+
+            reject("Message not received in a reasonable time");
+          }
+          counter++;
+        }, 10);
+      });
+    }
+    return mm.msg_data;
+  };
+}
+
+if (globalThis.js_send_msg === undefined) {
+  globalThis.js_send_msg = async function (session: string, self_index: number, party: number, msg_type: string, msg_data?: string) {
+    const tss_client = globalThis.tss_clients.get(session) as Client;
+    tss_client.log(`sending msg, ${msg_type}`);
+    if (msg_type.indexOf("ga1_data_unprocessed") > -1) {
+      throw new Error("ga1_data_unprocessed should not be sent directly");
+    }
+
+    if (tss_client.websocketOnly) {
+      const socket = tss_client.sockets[party];
+      socket.emit("send_msg", {
+        session,
+        sender: self_index,
+        recipient: party,
+        msg_type,
+        msg_data,
+      });
+    } else {
+      const sid = session.split(DELIMITERS.Delimiter4)[1];
+      const endpoint = tss_client.lookupEndpoint(session, party);
+      fetch(`${endpoint}/send`, {
+        method: "POST",
+        headers: {
+          [WEB3_SESSION_HEADER_KEY]: sid,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          session,
+          sender: self_index,
+          recipient: party,
+          msg_type,
+          msg_data,
+        }),
+      });
+    }
+    return true;
+  };
+}
```

Comment:
did you just move this piece of code? or also modify?

Reply:
just moved, eslint was complaining because it was being used before its defined